### ISSUE TITLE
Help with BIOMOD_xxx - Hi, I am using biomod2 with binary data for future projections with GLM, GAM, GBM, RF but when the model gives error for GBM Error in serialize(data, node$con) : error writing to connection. Any idea why? The other methods work normally. 

### ISSUE THREAD

**Issue by agnesM90** [biomodhub/biomod2#348]
Here is my script. I used the same with  biomod2 4.1 and it was fine. After the update is not working anymore. 

Options <-BIOMOD_ModelingOptions(GLM = list( type = 'quadratic',
                                                 interaction.level = 0,
                                                 myFormula = NULL, 
                                                 test = 'AIC',
                                                 family = binomial(link = 'logit')),
                                     GAM = list( algo = 'GAM_mgcv',
                                                 k = 5, # Or k=5
                                                 type = 's_smoother',
                                                 interaction.level = 0,
                                                 test = 'GCV.Cp',
                                                 myFormula = NULL, 
                                                 family = binomial(link = 'logit')),
                                     GBM = list( distribution = 'bernoulli',
                                                 n.trees = 5000, 
                                                 interaction.depth = 5,
                                                 n.minobsinnode = 5,
                                                 shrinkage = 0.05,
                                                 bag.fraction = 0.75,
                                                 perf.method = 'OOB'),
                                     RF = list(  do.classif = FALSE,
                                                 ntree = 200,
                                                 mtry = 3, 
                                                 nodesize = 10,
                                                 maxnodes = NULL)
)

# using list of response names (random samples) you can perform multiple BIOMOD runs.
bins <- c("bin1", "bin2", "bin3", "bin4", "bin5", "bin6", "bin7", "bin8", "bin9", "bin10")

bins <- c("bin1") # for one run

for (bin in bins)  {
  myRespName<-bin
  myResp<-as.numeric(al_cal[,myRespName]) #Note, that column names and raster names need to match perfectly
  myRespXY<-al_cal[,c("x","y")]
  myExpl<-envs
  
  myEvalRespName<-bin
  myEvalResp<-as.numeric(al_eval[,myEvalRespName])
  myEvalXY<-al_eval[,c("x","y")]
  
  al_data <- BIOMOD_FormatingData(resp.var = myResp,
                                   expl.var = myExpl, 
                                   resp.xy = myRespXY, 
                                   resp.name = myRespName,
                                   eval.resp.var = myEvalResp,
                                   eval.expl.var = myExpl,
                                   eval.resp.xy = myEvalXY
  )
  

  AL_Eval_Out_bins <- BIOMOD_Modeling(bm.format = al_data,
                                       models = c('GAM', 'GLM','RF',  'GBM'),
                                       bm.options = Options,
                                       CV.nb.rep = 1, 
                                       CV.perc = 0.7, 
                                       var.import = 10, 
                                       metric.eval = c('ROC','TSS'),
                                       do.full.models=FALSE, 
                                       modeling.id=paste(myRespName,"_eval",sep=""))
  # Ensemble model
  AL_Eval_Ens_Out <- BIOMOD_EnsembleModeling(
    bm.mod = AL_Eval_Out_bins,
    models.chosen = 'all',
    em.by='all',
    em.algo = c('EMmean'),
    metric.select = c('TSS'),
    prob.mean = T,
    prob.cv = F,
    prob.ci = F,
    prob.ci.alpha = 0.05,
    prob.median = F,
    committee.averaging = F)
  
  options(max.print = 1000000)

**Comment by MayaGueguen** [comment]
Hello Agnes,

Could you please share with me the output of  the `sessionInfo()` command ? :pray: 
Definition of modeling options has changed in the last version, so it might be that you are using default options that are really not optimised for this context. But I need to know with which version you are working :) 

Maya

**Comment by agnesM90** [comment]
Hi,

Thanks for the quick response!
Here is the output 😊

R version 4.2.3 (2023-03-15 ucrt)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 19045)

Matrix products: default

locale:
[1] LC_COLLATE=English_United Kingdom.utf8  LC_CTYPE=English_United Kingdom.utf8    LC_MONETARY=English_United Kingdom.utf8 LC_NUMERIC=C
[5] LC_TIME=English_United Kingdom.utf8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base

other attached packages:
[1] biomod2_4.2-4 terra_1.7-23  raster_3.6-20 readxl_1.4.2  plyr_1.8.8    sp_1.6-0      gbm_2.1.8.1

loaded via a namespace (and not attached):
[1] tidyselect_1.2.0       reshape2_1.4.4         splines_4.2.3          lattice_0.20-45        colorspace_2.1-0       vctrs_0.6.2            generics_0.1.3
 [8] mgcv_1.9-0             utf8_1.2.3             survival_3.5-3         rlang_1.1.0            pillar_1.9.0           glue_1.6.2             xgboost_1.7.5.1
[15] plotmo_3.6.2           foreach_1.5.2          lifecycle_1.0.3        mda_0.5-4              stringr_1.5.0          munsell_0.5.0          gtable_0.3.3
[22] cellranger_1.1.0       codetools_0.2-19       maxnet_0.1.4           parallel_4.2.3         class_7.3-22           fansi_1.0.4            Rcpp_1.0.10
[29] scales_1.2.1           plotrix_3.8-2          jsonlite_1.8.7         abind_1.4-5            TeachingDemos_2.12     ggplot2_3.4.2          stringi_1.7.12
[36] dplyr_1.1.1            dismo_1.3-14           grid_4.2.3             rgdal_1.6-6            cli_3.6.1              tools_4.2.3            magrittr_2.0.3
[43] PresenceAbsence_1.1.11 tibble_3.2.1           randomForest_4.7-1.1   Formula_1.2-5          pkgconfig_2.0.3        MASS_7.3-60            Matrix_1.6-0
[50] data.table_1.14.8      pROC_1.18.0            reshape_0.8.9          rstudioapi_0.14        iterators_1.0.14       rpart_4.1.19           earth_5.3.2
[57] R6_2.5.1               nlme_3.1-162           nnet_7.3-19            compiler_4.2.3

**Comment by agnesM90** [comment]
Thanks! Yes, it does go through the end with GLM, GAM and RF just gives the error to GBM. I tried to restart my session and computer and I even tried with the older version but with no success.

**Comment by MayaGueguen** [comment]
Then I'm afraid it is linked to the GBM function and package, and the data you have...
If your data did not change, try and check if some updates have been made into the `gbm` package : https://github.com/gbm-developers/gbm

**Comment by Daviser95** [comment]
Hi! I am getting the same error when running the BIOMOD_Modeling with GLM, GBM, MARS and RF. All the models end but GBM fails. Moreover, using the same code, but a different set of variables, GBM did not fail and ended well, but after changin the variables with some at a finer resolution, GBM starts to fail. I suspect it is related to a memory issue, because checking the memory use during the function, when it comes to GBM all memory was used.
